splink/comparison_level_library.py

```diff
@@ -0,0 +1,51 @@
+from .comparison_level_creator import ComparisonLevelCreator
+from .dialects import dialect_lookup
+
+
+class NullLevel(ComparisonLevelCreator):
+    def create_sql(self, sql_dialect):
+        col = self.input_column(sql_dialect)
+        return f"{col.name_l()} IS NULL OR {col.name_r()} IS NULL"
+
+    def create_label_for_charts(self):
+        return f"{self.col_name} is NULL"
+
+
+class ElseLevel(ComparisonLevelCreator):
+    def create_sql(self, sql_dialect):
+        return "ELSE"
+
+    def create_label_for_charts(self):
+        return "All other comparisons"
+
+
+class ExactMatchLevel(ComparisonLevelCreator):
+    def __init__(self, col_name: str, term_frequency_adjustments: bool = True):
+        config = {}
+        if term_frequency_adjustments:
+            config["tf_adjustment_column"] = col_name
+            config["tf_adjustment_weight"] = 1.0
+            # leave tf_minimum_u_value as None
+        super().__init__(col_name)
+        self.configure(**config)
+
+    def create_sql(self, sql_dialect):
+        col = self.input_column(sql_dialect)
+        return f"{col.name_l()} = {col.name_r()}"
+
+    def create_label_for_charts(self):
+        return f"Exact match {self.col_name}"
+
+
+class LevenshteinLevel(ComparisonLevelCreator):
+    def __init__(self, col_name: str, distance_threshold: float):
+        self.distance_threshold = distance_threshold
+        super().__init__(col_name)
+
+    def create_sql(self, sql_dialect):
+        col = self.input_column(sql_dialect)
+        lev_fn = dialect_lookup[sql_dialect].levenshtein_function_name
+        return f"{lev_fn}({col.name_l()}, {col.name_r()}) <= {self.distance_threshold}"
+
```

Comment:
We could hide `dialect_lookup` from people implementing concrete levels using a trick [like this](https://github.com/ADBond/alternative-approaches/blob/1c8c4ca8e09e9e1c28be189c632b68ccacd0f756/comparison_level.py#L75-L90).
This is probably a bit simpler for contributers, having something like
```
lev_fn = self.dialect.levenshtein_function_name
```
without having to worry about the fact that the object doesn't _really_ have a dialect, but it comes at some added complexity to the base class

Reply:
they would also then have (potentially) direct access to things like `name_l` etc without needing to directly go via `InputColumn`